core-services/prow/02_config/openshift/builder/_prowconfig.yaml

```diff
@@ -54,7 +54,63 @@ tide:
     - master
     labels:
     - approved
+    - bugzilla/valid-bug
+    - lgtm
+    missingLabels:
+    - backports/unvalidated-commits
+    - bugzilla/invalid-bug
+    - do-not-merge/hold
+    - do-not-merge/invalid-owners-file
+    - do-not-merge/work-in-progress
+    - keep-main-query-separate
+    - needs-rebase
+    repos:
+    - openshift/builder
+  - excludedBranches:
+    - feature-es6x
+    - feature-prom-connector
+    - main
+    - master
```

Comment:
Are you sure this is what you want, you exclude both master/main and the various release branches?

Reply:
well, that's what the existing exclude block was doing.  I just made a copy of it with a different set of labels.

hence my statement that it seems redundant to the other block that explicitly targets the main/master branch, but uses the same query.

Reply:
see: https://github.com/openshift/release/pull/20282/files#diff-185e498b2c2f60a52d987d975331a885c2fe9af3f9a4804a1dee96511a7f5849L68

Reply:
But this is not what you described, you said:

> In particular it seems like the "exclude branches" query and the "include: master/main" query are duplicative.... 

This query targets all branches that are neither master main nor release branches, so it is not duplicate with a include: master/main query at all. I am not sure why it exists, maybe because ppl sometimes create random feature branches and want a distinct config for them?

Reply:
> so it is not duplicate with a include: master/main query at all

what i mean is, the two blocks both use the same query.  So there's no reason to have two different blocks.  If we removed main/master from the "exclude" block, then we would not need the other "include: main/master" block, because the net behavior would be the same.

so yes, maybe sometimes the configs are different and that's why they are separate.  At the moment, the configs are the same, thus there's no need for two different configs, they could share a common one.

I chose to leave it separate in the interest of touching as little as possible

Reply:
@bparees @alvaroaleman the reason why we were keeping separate queries is that merge criteria change over time for `master/main` (we need bugs in the feature freeze period and we don't outside of it). Keeping separate queries made that event a simple change ("add a required label to master/main query") and not a complicated one ("make a copy of the unified query, make one exlcude master, make one inlclude *just* master, add a required label in the latter").

Once we are able to fully maintain the configs with automation I'd expect it to not have the duplicates.